components/DeckCountTextureBind.cpp

```diff
@@ -38,18 +49,49 @@ void DeckCountTextureBind::start()
 		kitten::Event::EventType::Card_Discarded,
 		this,
 		std::bind(&DeckCountTextureBind::deckEventReceiver, this, std::placeholders::_1, std::placeholders::_2));
+	kitten::EventManager::getInstance()->addListener(
+		kitten::Event::EventType::Enemy_Draw_Card,
+		this,
+		std::bind(&DeckCountTextureBind::enemyDrawEventReceiver, this, std::placeholders::_1, std::placeholders::_2));
 
 	m_currentTexPair = m_texPairs.rbegin();
 	m_attachedObject->getComponent<userinterface::UIFrame>()->setTexture(m_currentTexPair->second.c_str());
 
-	kitten::K_GameObject* counter = kitten::K_GameObjectManager::getInstance()->createNewGameObject("ui/deck/deck_counter_textbox.json");
-	puppy::TextBox* txtBoxComp = counter->getComponent<puppy::TextBox>();
-	txtBoxComp->setText("Loading...");
 	const glm::vec2 deckScale = getTransform().getScale2D();
 	const glm::vec3 deckTrans = getTransform().getTranslation();
-	counter->getTransform().place2D(deckTrans.x + 20, deckTrans.y + deckScale.y + 10);
+
+	bool host = false;
+	if (networking::ClientGame::getClientId == 0)
```

Comment:
could have just put `bool host = (networking::ClientGame::getClientId == 0);` :p

Reply:
It isn't. Fixing.